src/org/anjocaido/groupmanager/dataholder/WorldDataHolder.java

```diff
@@ -525,23 +516,17 @@ public boolean purgeTimedPermissions() {
 
 		boolean expired = false;
 
-		synchronized (groups) {
-
-			for (Group group : getGroups().values()) {
-				if (group.removeExpired()) {
-					setGroupsChanged(true);
-					expired = true;
-				}
+		for (Group group : getGroups().values()) {
+			if (group.removeExpired()) {
+				setGroupsChanged(true);
+				expired = true;
 			}
 		}
 
-		synchronized (users) {
-
-			for (User user : getUsers().values()) {
-				if (user.removeExpired()) {
-					setUsersChanged(true);
-					expired = true;
-				}

```

Comment:
I don't have an environment where I can do prolonged and extensive testing, so either this would need to be reverted or it would be published as a potentially highly unstable build